projects/com.oracle.truffle.llvm.parser.bc/src/com/oracle/truffle/llvm/parser/bc/irwriter/ConstantPrintVisitor.java

```diff
@@ -174,14 +175,12 @@ public void visit(CompareConstant compareConstant) {
 
     @Override
     public void visit(DoubleConstant doubleConstant) {
-        final long bits = Double.doubleToRawLongBits(doubleConstant.getValue());
-        out.print(String.format("0x%x", bits));
+        out.print(String.format(Locale.ROOT, "%e", doubleConstant.getValue()));
```

Comment:
See http://llvm.org/releases/3.2/docs/LangRef.html#simpleconstants why we should use the Hex-Representation.